Eos/Soave-Redlich-Kwong/EOS.H

```diff
@@ -283,75 +321,73 @@ AMREX_FORCE_INLINE
 void
 TY2Cp(amrex::Real T, amrex::Real Y[], amrex::Real& Cp)
 {
-  amrex::Error("TY2Cp not physically possible for this EoS");
+  amrex::Error("TY2Cp not physically possible for SRK EoS");
```

Comment:
Yeah, I wondered what the best approach to this issue would be.  I'm ok with this strategy for now.  The alternative is that we retrofit all the EOS's with a consistent set of dependencies.

Reply:
The next step is to implement this in PeleC so I'll have to decide how to handle this. Option A is to add wrappers like RTY2Cp around TY2CP in other EoS. Option B is leave the interfaces as is and put ifdefs around the EoS calls in PeleC to pick the right interface at compile time. I see advantages and disadvantages to both. I think there is some advantage to having the function interface reflect the physics (Option B), but having a single interface for all EoS is more elegant from a programming perspective. I'd be happy to hear your opinion before starting down one road or the other.

Reply:
I'd say to start the process as you have.  If significant PeleC code needs to be re-orchestrated to make available the data for the enhanced dependency, then if-def'ing becomes impractical (or, at least difficult to read/debug).  The trade-off will then be based on the performance gains of the simpler interface vs. code readability and maintainability.

Reply:
Thanks for the thoughts. In pretty much all cases, the added dependencies for SRK should be available without added effort, so I'm leaning toward the ifdefing option. We'll see if the story changes when I actually sit down to do the implementation though.